docs/reference/glossary.md

```diff
@@ -181,22 +187,58 @@ labeled using [tags](#tag).
 Here is an example of the shared [nginx repository](https://hub.docker.com/_/nginx/)
 and its [tags](https://hub.docker.com/r/library/nginx/tags/)
 
-## Swarm
 
-[Swarm](https://github.com/docker/swarm) is a native clustering tool for Docker.
-Swarm pools together several Docker hosts and exposes them as a single virtual
-Docker host. It serves the standard Docker API, so any tool that already works
-with Docker can now transparently scale up to multiple hosts.
+## service
+
+A [service](../swarm/how-swarm-mode-works/services.md) is the definition of how
+you want to run your application containers in a swarm. At the most basic level
+a service  defines which container image to run in the swarm and which commands
+to run in the container. For orchestration purposes, the service defines the
+"desired state", meaning how many containers to run as tasks and where to deploy
+the containers.
+
+Frequently a service is a microservice within the context of some larger
+application. Examples of services might include an HTTP server, a database, or
+any other type of executable program that you wish to run in a distributed
+environment.
+
+## service discovery
+
+Swarm mode [service discovery](../swarm/networking.md) is a DNS component
+internal to the swarm that automatically assigns each service in the swarm a VIP
+and DNS entry.
```

Comment:
This is the technical explanation, but we're not explaining why you should care. What about

Service discovery is the ability of a service to find where other services can be reached and consumed, as the physical infrastructure keeps changing and services are moved from one node to another.

@thaJeztah how does this work? Services running in the same network can discover one another through <service-name>:<port>?